src/browser/ui/dom/DOMChildrenOperations.js

```diff
@@ -31,10 +31,21 @@ function insertChildAt(parentNode, childNode, index) {
   // rely exclusively on `insertBefore(node, null)` instead of also using
   // `appendChild(node)`. However, using `undefined` is not allowed by all
   // browsers so we must replace it with `null`.
-  parentNode.insertBefore(
-    childNode,
-    parentNode.childNodes[index] || null
-  );
+
+  // fix render order error in safari
+  if (!(document.all && !document.addEventListener)) {
```

Comment:
`if (parentNode.childNodes.item)` is an exact feature test, no need to pin-point a specific browser with a generic test. Although you might want to make it `document.childNodes.item` (if `id="item"` would interfere?). Anyway, I just tested in IE8 and `childNodes.item` is there, so I'm not sure if this fallback is even necessary?